apollo-backend/src/main/java/io/logz/apollo/controllers/ScalingController.java

```diff
@@ -82,6 +90,24 @@ public void updateScalingFactor(int groupId, int scalingFactor, Req req) {
         group.setScalingStatus(Group.ScalingStatus.PENDING);
         groupDao.updateGroup(group);
 
+        String userEmail = req.token(Tokens._USER).toString();
+        Service service = serviceDao.getService(group.getServiceId());
+        Environment environment = environmentDao.getEnvironment(group.getEnvironmentId());
+
+        MDCLogging.withMDCFields(ImmutableMap.of("userEmail", userEmail,
+                                                "groupName", group.getName(),
+                                                "serviceName", service.getName(),
+                                                "envName", environment.getName(),
+                                                "newScalingFactor", String.valueOf(scalingFactor)),
+            () -> logger.info("Scaling factor changed from {} to {} for group {} by user email {}, env ID: {}, service ID: {}, group ID: {}",
+                group.getScalingFactor(),
```

Comment:
`group.getScalingFactor()` will return an updated value because you have group.setScalingFactor(scalingFactor) before. Log will print the same value in `from` and `to`, no?

Reply:
@matvey-mtn is right :)
You can save the old scalingFactor before changing it

Reply:
Good catch!